src/client/pages/Offer/index.tsx

```diff
@@ -0,0 +1,221 @@
+import {
+  useHistory,
+  useRouteMatch,
+  RouteComponentProps,
+  Redirect,
+} from 'react-router'
+import { LoadingPage } from 'components/LoadingPage'
+import { TopBar } from 'components/TopBar'
+import { Page } from 'components/utils/Page'
+import { useQuoteCartQuery, QuoteBundleVariant } from 'data/graphql'
+import { useVariation, Variation } from 'utils/hooks/useVariation'
+import { trackOfferGTM, EventName } from 'utils/tracking/gtm'
+import { getUtmParamsFromCookie, TrackAction } from 'utils/tracking/tracking'
+import { localePathPattern } from 'l10n/localePathPattern'
+import { Features, useFeature } from 'utils/hooks/useFeature'
+import { useSelectedInsuranceTypes } from 'utils/hooks/useSelectedInsuranceTypes'
+import { useCurrentLocale } from 'l10n/useCurrentLocale'
+import { LanguagePicker } from '../Embark/LanguagePicker'
+import {
+  getOfferData,
+  getUniqueQuotesFromVariantList,
+  getBundleVariantFromInsuranceTypes,
+  getInsuranceTypesFromBundleVariant,
+} from '../OfferNew/utils'
+import { AppPromotionSection } from '../OfferNew/AppPromotionSection'
+import { Checkout } from '../OfferNew/Checkout'
+import { FaqSection } from '../OfferNew/FaqSection'
+import { Introduction } from '../OfferNew/Introduction'
+import { Perils } from '../OfferNew/Perils'
+import { InsuranceSelector } from '../OfferNew/InsuranceSelector'
+
+const createToggleCheckout = (history: History<any>, locale?: string) => (
+  isOpen: boolean,
+) => {
+  if (isOpen) {
+    history.push(`/${locale}/new-member/sign`)
+  } else {
+    history.goBack()
+  }
+}
+
+type OfferPageProps = RouteComponentProps<{ id: string }>
+
+export const OfferPage = ({
+  match: {
+    params: { id: quoteCartId },
+  },
+}: OfferPageProps) => {
+  const { isoLocale, path: pathLocale } = useCurrentLocale()
+  const variation = useVariation()
+  const [isInsuranceToggleEnabled, isQuoteCartEnabled] = useFeature([
+    Features.OFFER_PAGE_INSURANCE_TOGGLE,
+    Features.QUOTE_CART_API,
+  ])
+
+  const [
+    selectedInsuranceTypes,
+    setSelectedInsuranceTypes,
+  ] = useSelectedInsuranceTypes()
+
+  const history = useHistory()
+
+  const {
+    data: quoteCartData,
+    loading: isLoadingQuoteCart,
+    refetch: refetchQuoteCart,
+    error: quoteCartError,
+  } = useQuoteCartQuery({
+    variables: {
+      id: quoteCartId,
+      locale: isoLocale,
+    },
+  })
+
+  const bundleData = quoteCartData?.quoteCart.bundle
+  const redeemedCampaigns = [quoteCartData?.quoteCart.campaign]
+
+  if (quoteCartError) {
+    throw new Error(`Error fetching quote cart: ${quoteCartId}.`)
+  }
+
+  if (!isLoadingQuoteCart && !bundleData) {
+    throw new Error(
+      `No bundle returned to show offer with (quote cart ID=${quoteCartId}).`,
+    )
+  }
+
+  const bundleVariants = (bundleData?.possibleVariations ?? []) as Array<
+    QuoteBundleVariant
+  >
+
+  const isInsuranceSelectorVisible =
+    isInsuranceToggleEnabled && bundleVariants.length > 1
+
+  const selectedBundleVariant =
```

Comment:
It was my bad to use "contract type" instead of "insurance type". The new thing in Embark is that we will use insurance type to decide pre-selected bundle. Before we used quote IDs for the same purpose.